Improve/update Cloud Quickstart (#93)
astronomer/docs: cloud/README.md

Astronomer Cloud runs on a Kubernetes cluster hosted and managed by the Astronomer team. Our managed offering abstracts the everyday data engineer from the infrastructure management layer otherwise required to adopt, maintain and grow with Airflow.

## Get Started

> If you're interested in giving Astronomer Cloud a spin, you can start a free 14-day trial by [reaching out to us here](/get-astronomer?ref=docs). is the best way to get started. You're welcome to sign up at anytime, no credit card required.

Reviewer: ```suggestion
If you're interested in trying out Astronomer Cloud, start a free 14-day trial by [reaching out to us here](/get-astronomer?ref=docs). You're welcome to sign up at anytime, no credit card required.
```

Reviewer: ("Giving Astronomer Cloud a spin" feels like too much like lingo/not accessible, so simplified it)

Reviewer: Ok I fixed this :)
